Warning in MSVC about implicit casting

From #161

```
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include\utility(248,16): warning C4244: 'initializing': conversion from '_Ty' to '_Ty1', possible loss of data
1>        with
1>        [
1>            _Ty=double
1>        ]
1>        and
1>        [
1>            _Ty1=float
1>        ] (compiling source file ..\src\player.cpp)
1>C:\Users\V\Documents\GitHub\Cataclysm-BN\src\player.cpp(467): message : see reference to function template instantiation 'std::pair<float,float>::pair<double,float,0>(std::pair<double,float> &&) noexcept' being compiled
1>C:\Users\V\Documents\GitHub\Cataclysm-BN\src\player.cpp(461): message : see reference to function template instantiation 'std::pair<float,float>::pair<double,float,0>(std::pair<double,float> &&) noexcept' being compiled
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.27.29110\include\utility(248,16): warning C4244: 'initializing': conversion from '_Ty' to '_Ty1', possible loss of data
1>        with
1>        [
1>            _Ty=int
1>        ]
1>        and
1>        [
1>            _Ty1=float
1>        ] (compiling source file ..\src\player.cpp)
1>C:\Users\V\Documents\GitHub\Cataclysm-BN\src\player.cpp(485): message : see reference to function template instantiation 'std::pair<float,float>::pair<int,float,0>(std::pair<int,float> &&) noexcept' being compiled
1>C:\Users\V\Documents\GitHub\Cataclysm-BN\src\player.cpp(479): message : see reference to function template instantiation 'std::pair<float,float>::pair<int,float,0>(std::pair<int,float> &&) noexcept' being compiled
```

--- a/src/player.cpp
+++ b/src/player.cpp
@@ -461,7 +461,7 @@ int player::kcal_speed_penalty() const
     static const std::vector<std::pair<float, float>> starv_thresholds = { {
             std::make_pair( 0.0f, -90.0f ),
             std::make_pair( 0.1f, -50.f ),
-            std::make_pair( 0.3, -25.0f ),
+            std::make_pair( 0.3f, -25.0f ),
             std::make_pair( 0.5f, 0.0f )
         }
     };
@@ -477,10 +477,10 @@ int player::thirst_speed_penalty( int thirst )
     // We die at 1200 thirst
     // Start by dropping speed really fast, but then level it off a bit
     static const std::vector<std::pair<float, float>> thirst_thresholds = {{
-            std::make_pair( +thirst_levels::very_thirsty, 0.0f ),
-            std::make_pair( +thirst_levels::dehydrated, -25.0f ),
-            std::make_pair( +thirst_levels::parched, -50.0f ),
-            std::make_pair( +thirst_levels::dead, -75.0f )
+            std::make_pair( static_cast<float>( thirst_levels::very_thirsty ), 0.0f ),
+            std::make_pair( static_cast<float>( thirst_levels::dehydrated ), -25.0f ),
+            std::make_pair( static_cast<float>( thirst_levels::parched ), -50.0f ),
+            std::make_pair( static_cast<float>( thirst_levels::dead ), -75.0f )
         }
     };
     return static_cast<int>( multi_lerp( thirst_thresholds, thirst ) );

